scripts/airdropBsxToHdx/getData.ts

```diff
@@ -0,0 +1,113 @@
+// ✅ -> discard treasury and intergalactic acc
+    // 0x6d6f646c70792f74727372790000000000000000000000000000000000000000 (HDX Treasury)
+    // 0x0abad795adcb5dee45d29528005b1f78d55fc170844babde88df84016c6cd14d (Founders & Investors)
+// ✅ for an individual account sum free and reserved balance 💪
+// ✅ total all balances for use in calculating airdrop according to weight
+
+// ✅ create an 'accounts' .json ?  probably.. like 'contribs' for documentation purposes :| :)
+
+
+const { ApiPromise, WsProvider, Keyring } = require("@polkadot/api")
+const { deriveAddress, encodeAddress, cryptoWaitReady } = require("@polkadot/util-crypto")
+const { stringToU8a } = require("@polkadot/util")
+const BN = require("BN.js")
+
+interface AirdropBsxToHdxData {
+    totalAllOgHdxBalances: string
+    OgAccounts: OgAccount[]
+}
+
+interface OgAccount {
+    address: string
+    freeBalance: string
+}
+
+const hdxTreasuryPubKey = '0x6d6f646c70792f74727372790000000000000000000000000000000000000000'
+const foundersInvestorsPubKey = '0x0abad795adcb5dee45d29528005b1f78d55fc170844babde88df84016c6cd14d'
+
+async function main() {
+    await cryptoWaitReady()
+    const hdxArchiveNode = 'wss://archive.snakenet.hydradx.io'
+    const provider = new WsProvider(hdxArchiveNode)
+    const api = await ApiPromise.create({provider});
+
+    // Get general information about the node we are connected to
+    const [chain, nodeName, nodeVersion] = await Promise.all([
+        api.rpc.system.chain(),
+        api.rpc.system.name(),
+        api.rpc.system.version()
+    ]);
+    console.log(
+        `
+        \n You are connected to ${chain}
+        \n via ${hdxArchiveNode}
+        \n using ${nodeName} v${nodeVersion} \n
+        `
+    )
+
+    let totalAllOgHdxBalances = new BN('0')
+    const blockHeight = '1999999'
+    //   ^^^  block height at which to get orignial HDX accounts
+    //   link or explanation for this blockheight ?
+    const treasuryHdxAcct = hdxAddress(hdxTreasuryPubKey)
+    const foundersInvestorsHdxAcct = hdxAddress(foundersInvestorsPubKey)
+    const ogHdxAccts: OgAccount[] = await api.rpc.chain.getBlockHash(blockHeight)
+            .then( blockHash => api.rpc.chain.getBlock(blockHash) )
+            .then( signedBlock => signedBlock.block.header.hash.toHex() )
+            .then( blockHex => api.query.system.account.entriesAt(blockHex) )
+            .then( accountsData => accountsData.reduce(( acctsWeWant, [key, {data}] ) => {
+                const [address] = key.toHuman()
+                if (
+                    address === treasuryHdxAcct
+                    || address === foundersInvestorsHdxAcct
+                ) {
+                    console.log('no airdrop for: ', address)
+                    return acctsWeWant // we return the accounts we want _without_ adding treasury or founders/investors
+                } else {
+                    const hdxBalanceTotal = new BN(data.free.toString())
```

Comment:
There is the [minimum amount](https://paritytech.github.io/substrate/latest/pallet_vesting/pallet/trait.Config.html#associatedtype.MinVestedTransfer) that can be vested and is set to [100 000](https://github.com/galacticcouncil/Basilisk-node/blob/11a6798d567a0941337a33912b404c16da3a7480/runtime/common/src/lib.rs#L263). Maybe we should also check that all balances are higher, otherwise we risk [early interruption](https://github.com/paritytech/substrate/blob/3aecef7ee350db9cf53369481ebd1cee26ecc7ff/frame/utility/src/lib.rs#L113) of a batch call.